typed/clj.analyzer/src/typed/clj/analyzer/utils.cljc

```diff
@@ -499,9 +499,12 @@
   "Given a vector of arg tags and a collection of methods, tries to return the
    subset of methods that match best the given tags"
   [tags methods]
-  (let [o-tags (mapv #(or (maybe-class %) Object) tags)]
-    (if-let [methods (or (seq (filterv #(= o-tags (mapv maybe-class (:parameter-types %))) methods))
-                         (seq (filterv #(tag-match? tags %) methods)))]
+  (let [o-tags (mapv #(or (maybe-class %) Object) tags)
+        methods' (filterv #(= o-tags (mapv maybe-class (:parameter-types %)))
+                          methods)
+        methods' (if (seq methods') methods'
+                     (filterv #(tag-match? tags %) methods))]
+    (if (seq methods')
```

Comment:
I think we discussed this was because you wanted to iterate over the vector instead of the `seq`. What about:

```clojure
(or (not-empty (filterv ...))
    (not-empty (filterv ...)))
```
```

Reply:
Right, good idea.